File: Vanilla/sources/zdoc/media/lua/shared/Library/PlayerFallingState.lua
---@class PlayerFallingState : zombie.ai.states.PlayerFallingState
---@field private _instance PlayerFallingState
PlayerFallingState = {}

---@public
---@param arg0 IsoGameCharacter
---@return void
function PlayerFallingState:execute(arg0) end

---@public
---@return PlayerFallingState
function PlayerFallingState:instance() end

---@public
---@param arg0 IsoGameCharacter
---@return void
function PlayerFallingState:exit(arg0) end

---@public
---@param arg0 IsoGameCharacter
---@return void
function PlayerFallingState:enter(arg0) end

File: Vanilla/sources/zomboid/zombie/ai/states/PlayerFallingState.java
package zombie.ai.states;

import zombie.ai.State;
import zombie.characters.IsoGameCharacter;

public final class PlayerFallingState extends State {
   private static final PlayerFallingState _instance = new PlayerFallingState();

   public static PlayerFallingState instance() {
      return _instance;
   }

   public void enter(IsoGameCharacter var1) {
      var1.setVariable("bHardFall", false);
      var1.clearVariable("bLandAnimFinished");
   }

   public void execute(IsoGameCharacter var1) {
   }

   public void exit(IsoGameCharacter var1) {
      var1.clearVariable("bHardFall");
      var1.clearVariable("bLandAnimFinished");
   }
}
